melodies_monet/driver.py

```diff
@@ -2520,13 +2870,26 @@ def plotting(self):
                             # First for p_index = 0 create the obs box plot data array.
                             
                             if p_index == 0:
-                                comb_bx, label_bx,region_bx = splots.calculate_multi_boxplot(pairdf, pairdf_reg,region_name=region_name, column=obsvar, 
-                                                                             label=p.obs, plot_dict=obs_dict)
+                                comb_bx, label_bx,region_bx = splots.calculate_multi_boxplot(pairdf,
+                                                                                             pairdf_reg,
+                                                                                             region_name=region_name, 
+                                                                                             interval_var=interval_var,
+                                                                                             interval_list=interval_list,
+                                                                                             column=obsvar, 
+                                                                                             label=p.obs,
+                                                                                             plot_dict=obs_dict)
                                 
                             # Then add the models to this dataarray.
-                            comb_bx, label_bx,region_bx = splots.calculate_multi_boxplot(pairdf, pairdf_reg, region_name= region_name,column=modvar, label=p.model, 
-                                                                         plot_dict=plot_dict, comb_bx=comb_bx,
-                                                                         label_bx=label_bx)
+                            comb_bx, label_bx,region_bx = splots.calculate_multi_boxplot(pairdf, 
+                                                                                         pairdf_reg, 
+                                                                                         region_name=region_name,
+                                                                                         interval_var=interval_var,
+                                                                                         interval_list=interval_list,
+                                                                                         column=modvar, 
+                                                                                         label=p.model, 
+                                                                                         plot_dict=plot_dict, 
+                                                                                         comb_bx=comb_bx,
+                                                                                         label_bx=label_bx)
                             
                             # For the last p_index make the plot.
```

Comment:
It would be nice to add an x-label to the multi-box plot when plotting the intervals. Something like "Binned by Observational Variable: [interval_var]"

To do this. interval_var is already set to the variable or None so you can just add an xlabel = interval_var below under splots.make_multi_boxplot() and then it will always be available for your plot when interval_labels is available and you can use it for making the xlabel for the make_multi_boxplot 

Let me know if you have questions on this.